hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/storage/HoodieHFileWriter.java

```diff
@@ -77,6 +81,8 @@ public HoodieHFileWriter(String instantTime, Path file, HoodieHFileConfig hfileC
     this.file = HoodieWrapperFileSystem.convertToHoodiePath(file, conf);
     this.fs = (HoodieWrapperFileSystem) this.file.getFileSystem(conf);
     this.hfileConfig = hfileConfig;
+    this.schema = schema;
+    this.schemaRecordKeyField = Option.ofNullable(schema.getField(hfileConfig.getKeyFieldName()));
```

Comment:
Can we simply use HoodieHFileReader.KEY_FIELD_NAME here instead of plugging in through hfileConfig? 

hFileConfig value is anyways being hardcoded to HoodieHFileReader.KEY_FIELD_NAME so this does not provide nay benefit. Furthermore, in the PR to actually fix the interfaces to provide the key-name-config to readers and writers, we will have greater flexibility in deciding how to pass the configs and refactor.

Reply:
Filed https://issues.apache.org/jira/browse/HUDI-3321 to track the config access to the writer/reader layers.